setup.py

```diff
@@ -66,6 +66,7 @@
     ext._add_ldflags(["-L", lib_path_npymath])  # pylint: disable=protected-access
     ext._add_ldflags(["-lnpymath"])  # pylint: disable=protected-access
     ext._add_cflags(["-I", np.get_include()])  # pylint: disable=protected-access
+    ext._add_cflags(["-std=c++11"])  # pylint: disable=protected-access
```

Comment:
Why is this required?

Reply:
This is necessary on older versions of clang that do not compile with c++11 by default. The include files of the wrapper (which add the Numpy header files), will use `constexpr` which was added in c++11.

Reply:
Why not supporting a more up-to-date standard, c++17 or c++20?

Reply:
Correct me if I'm wrong, but specifying the minimum version necessary for a piece of code to compile seems appropriate. If the code only uses c++-11 features, than there is no need to be more restrictive by requiring a compiler with c++-20 support.

Reply:
This is the bare minimum. [`clang` 16 switched to use c++17 by default](https://clang.llvm.org/cxx_status.html#:~:text=By%20default%2C%20Clang%2016%20or,in%20Clang%204%20and%20earlier). The `clang` binary shipped on macOS is `clang` 14. I just doubled checked, and it is pybind headers (included from Numpy headers) which use constexpr. [pybind11 is compatible with `c++11`.](https://pybind11.readthedocs.io/en/stable/compiling.html#configuration-variables) So I think we should keep `c++11`

Reply:
My point here is with consistency and c++11 is a pretty old standard. We should ensure the code builds with either C++17 or C++20 support. There isn't any  issue with updating this to C++17.
  ```suggestion
    ext._add_cflags(["-std=c++17"])  # pylint: disable=protected-access
```

Reply:
it is a small change and of course this's possible to compile separate targets with different standards, but it is best to be consistent. We agreed C++17 to be the minimum C++ standard and I think we can keep it as is across the project.

Reply:
Ok, we had agreed in the past about c++17, that's a good enough reason for me.

Reply:
Generally I agree with Erick. But if you insist on a consistent standard that's fine by me too. Note though that you already specify C++20 in the runtime, not 17.